File: streamCodeGenerator/src/main/java/com/manuga/streamCodeGenerator/helpers/databases/dbModels/DatabaseSettings.java
package com.manuga.streamCodeGenerator.helpers.databases.dbModels;

import com.manuga.streamCodeGenerator.helpers.helper.GenericHelper;

public class DatabaseSettings {
	
	private DatabaseTypeEnum dataBaseType = DatabaseTypeEnum.postgres;
	
	private String login;
	private String pass;
	private int port = 0;
	private String host;
	private String dataBase;
	private String schema;
	
	public DatabaseTypeEnum getDataBaseType() {
		return dataBaseType;
	}
	public void setDataBaseType(DatabaseTypeEnum dataBaseType) {
		this.dataBaseType = dataBaseType;
	}
	
	public String getLogin() {
		return login;
	}
	public void setLogin(String login) {
		this.login = login;
	}
	
	public String getPass() {
		return pass;
	}
	public void setPass(String pass) {
		this.pass = pass;
	}
	
	public int getPort() {		
		if(port == 0) {
			port = dataBaseType.getDefaultPort();
		}		
		return port;
	}
	public void setPort(int port) {
		this.port = port;
	}
	
	public String getHost() {
		return host;
	}
	public void setHost(String host) {
		this.host = host;
	}
	
	public String getDataBase() {
		return dataBase;
	}
	public void setDataBase(String dataBase) {
		this.dataBase = dataBase;
	}	
	
	
	
	public void saveToFile(String fName) throws Exception{
		GenericHelper.object2xml(this, fName);
	}

	public void loadFromFile(String fName) throws Exception{
		DatabaseSettings dbs = new DatabaseSettings();
		dbs = (DatabaseSettings) GenericHelper.xml2object(fName);
		this.login = dbs.login;
		this.pass = dbs.pass;
		this.port = dbs.port;
		this.host = dbs.host;
		this.dataBase = dbs.dataBase;
	}

    public String getSchema() {
        return schema;
    }

    public void setSchema(String schema) {
        this.schema = schema;
    }
}

File: simpleWebGenerator/classes/DatabaseSettings.php
<?php

class DatabaseSettings {

    private $dataBaseType = "postgres";
    private $login;
    private $pass;
    private $port = 0;
    private $host;
    private $dataBase;
    private $schema;

    public function getDataBaseType() {
        return $this->dataBaseType;
    }

    public function setDataBaseType($dataBaseType) {
        $this->dataBaseType = $dataBaseType;
    }

    public function getLogin() {
        return $this->login;
    }

    public function setLogin($login) {
        $this->login = $login;
    }

    public function getPass() {
        return $this->pass;
    }

    public function setPass($pass) {
        $this->pass = $pass;
    }

    public function getPort() {
        if ($this->port == 0) {
            $this->port = $this->getDataBaseType()->getDefaultPort();
        }

        return $this->port;
    }

    public function setPort($port) {
        $this->port = $port;
    }

    public function getHost() {
        return $this->host;
    }

    public function setHost($host) {
        $this->host = $host;
    }

    public function getDataBase() {
        return $this->dataBase;
    }

    public function setDataBase($dataBase) {
        $this->dataBase = $dataBase;
    }

    public function saveToFile($fName) {
        file_put_contents($fName, serialize($this));
    }

    public function loadFromFile($fName) {
        $dbs = unserialize(file_get_contents($fName));
        $this->login = $dbs->login;
        $this->pass = $dbs->pass;
        $this->port = $dbs->port;
        $this->host = $dbs->host;
        $this->dataBase = $dbs->dataBase;
    }

    public function getSchema() {
        return $this->schema;
    }

    public function setSchema($schema) {
        $this->schema = $schema;
    }

}
